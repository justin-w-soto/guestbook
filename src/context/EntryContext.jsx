import React from 'react'
import { useContext, createContext, useState } from 'react'

const EntryContext = createContext()

const EntryProvider = ({ children }) => {
  const [entries, setEntries] = useState([])

    return ( <EntryContext.Provider value={{ entries, setEntries }}>{ children }</EntryContext.Provider>)
}

const useEntries = () => {
  
  const context = useContext(EntryContext) 

  if (context === undefined) {
    throw new Error('useEntries must be inside of EntryContext Provider')
  }

  return context
}

export { EntryProvider, useEntries }
