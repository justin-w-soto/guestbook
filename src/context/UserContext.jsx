import React from 'react'
import { useContext, createContext, useState }from 'react'

    const UserContext = createContext()

    const UserProvider = ({ children }) => {
        const [user, setUser] = useState('')
        return (
        <UserContext.Provider 
        value={{user, setUser}}
        >
            { children }
        </UserContext.Provider>)
    }

    // MAKE A CUSTOM HOOK 
    const useMyHook = () => {
        const context = useContext(UserContext)
        
        // if (context === undefined) {
        //     throw new Error('useMyHook needs to be called inside of a UserContext Provider')
        // }
        return context
    }

   
export { UserProvider, useMyHook }
