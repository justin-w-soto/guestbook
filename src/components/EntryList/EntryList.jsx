import React from 'react'
import { useEntriesCustom } from '../../context/EntryContext'
import { Entry } from '../Entry/Entry'

export const EntryList = () => {
const { entries } = useEntriesCustom()

    return (
        <div>
            <ul>
                {entries.map((entry) => {
                    return (
                    <li key={`${entry.name}-${entry.message}`}> 
                        <Entry entry={entry}/>
                    </li>
                    )
                })}
            </ul>
        </div>
    )
}
