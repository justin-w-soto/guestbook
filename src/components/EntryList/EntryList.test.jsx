import React from "react";
import { render } from '@testing-library/react';
import { EntryList } from "./EntryList";
import { EntryProvider } from "../../context/EntryContext";

it('should render the notes', () => {
    const { container } = render(
    <EntryProvider>
        <EntryList />
    </EntryProvider> 
       )
    ;

    expect(container).toMatchSnapshot();
})