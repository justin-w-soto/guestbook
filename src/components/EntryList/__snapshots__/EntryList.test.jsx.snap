// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the notes 1`] = `
<div>
  <div>
    <ul />
  </div>
</div>
`;
